sdk/communication/AzureCommunication/Source/Identifiers.swift

```diff
@@ -77,28 +77,65 @@ import Foundation
  */
 @objcMembers public class PhoneNumberIdentifier: NSObject, CommunicationIdentifier {
     public let phoneNumber: String
+    public let id: String?
+
     /**
      Creates a PhoneNumberIdentifier object
      - Parameter phoneNumber: phone number to create the object, different from identifier
+     - Parameter id: Full id of the phone number
      */
-    public init(phoneNumber: String) {
+    public init(phoneNumber: String, id: String? = nil) {
         self.phoneNumber = phoneNumber
+        self.id = id
+    }
+    
+    public static func == (lhs: PhoneNumberIdentifier, rhs: PhoneNumberIdentifier) -> Bool {
+        if lhs.phoneNumber != rhs.phoneNumber {
+            return false
+        }
+        
+        return lhs.id == nil
+            || rhs.id == nil
+            || lhs.id == rhs.id;
     }
 }
 
 /**
  Communication identifier for Microsoft Teams Users
  */
 @objcMembers public class MicrosoftTeamsUserIdentifier: NSObject, CommunicationIdentifier {
+    public let id: String?
```

Comment:
I think we should avoid the use of names like `id` since `id` is a reserved type in Objective-C

Reply:
In most cases we use `public let identifier: String`